tasks/chain/economics.go

```diff
@@ -134,27 +141,38 @@ func (p *ChainEconomics) processItem(ctx context.Context, node lens.API, item *v
 		return xerrors.Errorf("get tipset: %w", err)
 	}
 
-	supply, err := node.StateVMCirculatingSupplyInternal(ctx, tsk)
+	ce, err := ExtractChainEconomicsModel(ctx, node, ts)
 	if err != nil {
-		return err
-	}
-
-	ce := &chainmodel.ChainEconomics{
-		ParentStateRoot: ts.ParentState().String(),
-		VestedFil:       supply.FilVested.String(),
-		MinedFil:        supply.FilMined.String(),
-		BurntFil:        supply.FilBurnt.String(),
-		LockedFil:       supply.FilLocked.String(),
-		CirculatingFil:  supply.FilCirculating.String(),
+		return xerrors.Errorf("extracting chain economics model: %w", err)
 	}
 
-	log.Debugw("persisting tipset", "height", int64(ts.Height()))
+	log.Debugw("persisting tipset", "height", item.Height)
 
-	if err := p.storage.DB.RunInTransaction(ctx, func(tx *pg.Tx) error {
+	if err := p.storage.RunInTransaction(ctx, func(tx *pg.Tx) error {
 		return ce.PersistWithTx(ctx, tx)
 	}); err != nil {
 		return xerrors.Errorf("persist: %w", err)
 	}
 
 	return nil
 }
+
+type ChainEconomicsLens interface {
```

Comment:
maybe disambiguate this from the current definition of lens used in sentinel. This is the subset of the API interface used, not an implementation of a backend.